src/main/java/org/apache/sling/resourceresolver/impl/mapping/MapEntries.java

```diff
@@ -179,7 +188,12 @@ public MapEntries(final MapConfigurationProvider factory,
         this.aliasMapsMap = new ConcurrentHashMap<>();
         this.stringInterpolationProvider = stringInterpolationProvider;
 
+        this.conflictingAliases = new ConcurrentLinkedQueue<>();
+        this.invalidAliases = new ConcurrentLinkedQueue<>();
```

Comment:
Do `this.conflictingAliases` and `this.invalidAliases` really need to be fields and concurret collections? An approach where they would be used on a single thread and within a single method (or passed around methods) would IMO make it much easier to reason about the concurrency model of this class.

Reply:
It's likely possible but might be more intrusive, because it might affect multiple method signatures. We already have a mix of of fields and parameters, for instance resolveMapsMap.

MapEntries already is convoluted and really needs a refactoring...

Re concurrency: you are right, this doesn't appear to be needed.

Reply:
But point taken, please check again now.